internal/common/cloudproviderapi/client.go

```diff
@@ -0,0 +1,163 @@
+package cloudproviderapi
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type Client struct {
+	authToken   string
+	apiURL      url.URL
+	client      *http.Client
+	retryConfig retryConfig
+}
+
+type retryConfig struct {
+	maxAttempts int
+	baseTimeout time.Duration
+	maxTimeout  time.Duration
+}
+
+func NewClient(authToken string, rawAPIURL string) (*Client, error) {
+	parsedAPIURL, err := url.Parse(rawAPIURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse Cloud Provider API url: %w", err)
+	}
+
+	return &Client{
+		authToken: authToken,
+		apiURL:    *parsedAPIURL,
+		client:    http.DefaultClient,
+		retryConfig: retryConfig{
+			maxAttempts: 5,
+			baseTimeout: 5 * time.Second,
+			maxTimeout:  125 * time.Second,
+		},
```

Comment:
I wonder if we should be doing our own custom retry solution as there looks to be an OOTB solution all the other clients use that works based off of provider configuration https://github.com/grafana/terraform-provider-grafana/blob/feacc9ffdb67ef4c695d9f0e850176c80507cc4d/pkg/provider/configure_clients.go#L274-L282

Reply:
ooh this is a great catch! I missed this, definitely would rather use the OOTB solution. Thank you!